packages/material-ui/src/Slider/Slider.js

```diff
@@ -1,1055 +1,294 @@
     cursor: 'pointer',
-    // Disable scroll capabilities.
     touchAction: 'none',
-    color: theme.palette.primary.main,
+    color: props.theme.palette.primary.main,
     WebkitTapHighlightColor: 'transparent',
-    '&$disabled': {
+    ...(props.state.color === 'secondary' && {
+      color: props.theme.palette.secondary.main,
+    }),
+    '&.Mui-disabled': {
       pointerEvents: 'none',
       cursor: 'default',
-      color: theme.palette.grey[400],
+      color: props.theme.palette.grey[400],
     },
-    '&$vertical': {
+    ...(props.state.orientation === 'vertical' && {
       width: 2,
       height: '100%',
       padding: '0 13px',
-    },
+    }),
     // The primary input mechanism of the device includes a pointing device of limited accuracy.
     '@media (pointer: coarse)': {
       // Reach 42px touch target, about ~8mm on screen.
       padding: '20px 0',
-      '&$vertical': {
+      ...(props.state.orientation === 'vertical' && {
         padding: '0 20px',
-      },
+      }),
     },
     '@media print': {
       colorAdjust: 'exact',
     },
-  },
-  /* Styles applied to the root element if `color="primary"`. */
-  colorPrimary: {
-    // TODO v5: move the style here
-  },
-  /* Styles applied to the root element if `color="secondary"`. */
-  colorSecondary: {
-    color: theme.palette.secondary.main,
-  },
-  /* Styles applied to the root element if `marks` is provided with at least one label. */
-  marked: {
-    marginBottom: 20,
-    '&$vertical': {
-      marginBottom: 'auto',
-      marginRight: 20,
-    },
-  },
-  /* Pseudo-class applied to the root element if `orientation="vertical"`. */
-  vertical: {},
-  /* Pseudo-class applied to the root and thumb element if `disabled={true}`. */
-  disabled: {},
-  /* Styles applied to the rail element. */
-  rail: {
-    display: 'block',
-    position: 'absolute',
-    width: '100%',
-    height: 2,
-    borderRadius: 1,
-    backgroundColor: 'currentColor',
-    opacity: 0.38,
-    '$vertical &': {
-      height: '100%',
-      width: 2,
-    },
-  },
-  /* Styles applied to the track element. */
-  track: {
-    display: 'block',
-    position: 'absolute',
-    height: 2,
-    borderRadius: 1,
-    backgroundColor: 'currentColor',
-    '$vertical &': {
-      width: 2,
-    },
-  },
-  /* Styles applied to the track element if `track={false}`. */
-  trackFalse: {
-    '& $track': {
-      display: 'none',
-    },
-  },
-  /* Styles applied to the track element if `track="inverted"`. */
-  trackInverted: {
-    '& $track': {
-      backgroundColor:
-        // Same logic as the LinearProgress track color
-        theme.palette.type === 'light'
-          ? lighten(theme.palette.primary.main, 0.62)
-          : darken(theme.palette.primary.main, 0.5),
+    ...(props.state.marked && {
+      marginBottom: 20,
+      ...(props.state.orientation === 'vertical' && {
+        marginBottom: 'auto',
+        marginRight: 20,
+      }),
+    }),
+
+    '& .MuiSlider-rail': {
```

Comment:
Backporting https://github.com/mnajdova/material-ui/pull/5#discussion_r485656210 in this pull request.

<img width="794" alt="Capture d’écran 2020-09-11 à 12 24 35" src="https://user-images.githubusercontent.com/3165635/92912126-ca1dab80-f429-11ea-93b6-1fb186febacc.png">

Reply:
We came to the conclusion to keep pushing in the current direction. While it increases specificity, it could help future customizations that start from the root element. Global class names are easier to understand than hashed ones. In the event this doesn't turn into a successful experimentation, we cost to going to a flat specificity is relatively acceptable. It doesn't require changing the base component.